File: backend/src/main/scala/microservices/attraction/objects/UploadAttractionImagePlannerRequest.scala
package com.typesafe.travel.attraction.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UploadAttractionImagePlannerRequest(
    originalFileName: String,
    mimeType: String,
    fileContentBase64: String
)
object UploadAttractionImagePlannerRequest:
  given sourceEncoder: Encoder[UploadAttractionImagePlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[UploadAttractionImagePlannerRequest] = deriveDecoder

File: frontend/src/microservices/operations/attraction/objects/UploadAttractionImagePlannerRequest.ts
// 本文件定义 attraction 模块的 `UploadAttractionImagePlannerRequest`，作为图片上传参数并提供 JSON 编解码。

export type UploadAttractionImagePlannerRequest = {
  originalFileName: string
  mimeType: string
  fileContentBase64: string
}

export const uploadAttractionImagePlannerRequestFromJson = (json: string): UploadAttractionImagePlannerRequest =>
  JSON.parse(json) as UploadAttractionImagePlannerRequest

export const uploadAttractionImagePlannerRequestToJson = (value: UploadAttractionImagePlannerRequest): string =>
  JSON.stringify(value)
